be/src/runtime/lake_tablets_channel.cpp

```diff
@@ -263,6 +265,12 @@ void LakeTabletsChannel::add_chunk(brpc::Controller* cntl, const PTabletWriterAd
                     count_down_latch.count_down();
                     continue;
                 }
+                // This AsyncDeltaWriter may have not been `open()`ed
+                if (auto st = dw->open(); !st.ok()) {
```

Comment:
Why not open when creating writer?

Reply:
Some partitions may not have data imported. For Tablets under these partitions, delaying opening can reduce memory usage because these Tablets don't need to `finish()`, as you can see from the above codes.